src/gpu/data/matrix.cu

```diff
@@ -43,6 +73,15 @@ namespace tsvd
 		);
 	}
 
+	void multiply(Matrix<tsvd_double>& A, const tsvd_double a, DeviceContext& context)
```

Comment:
Could these `multiply` methods be generalized into a single templated method, something like:

```
template <typename T>
void multiply(Matrix<T>& A, const T a, DeviceContext& context)
	{
		thrust::transform(A.dptr(), A.dptr() + A.size(), A.dptr(), [=]__device__ (T val)
		                  {
			                  return val * a;
		                  }
		);
	}
```
Or will the compiler complain?

Similar with other methods using `thrust` (add, subtract etc.) - anything not using `cublas`.

Reply:
You can do this it just needs to go in the header file not the .cu file.